tools/bwa.py

```diff
@@ -206,39 +206,35 @@ def align_mem_one_rg(self, inBam, refDb, outBam, rgid=None, options=None,
             #os.unlink(reheadered_bam)
 
     def mem(self, inReads, refDb, outAlign, options=None, min_score_to_filter=None,
-            threads=None):
+            threads=None, invert_filter=False):
         options = [] if not options else options
 
+        if '-t' not in options:
+            options.extend(('-t', str(threads)))
+
         threads = threads or util.misc.available_cpu_count()
         samtools = tools.samtools.SamtoolsTool()
-        fq1 = util.file.mkstempfname('.1.fastq')
-        fq2 = util.file.mkstempfname('.2.fastq')
-        aln_sam = util.file.mkstempfname('.sam')
-        samtools.bam2fq(inReads, fq1, fq2)
 
-        if '-t' not in options:
-            options.extend(('-t', str(threads)))
+        with util.file.tempfname('.aligned.sam') as aln_sam:
+            with samtools.bam2fq_tmp(inReads) as (fq1, fq2):
+                self.execute('mem', options + [refDb, fq1, fq2], stdout=aln_sam)
 
-        self.execute('mem', options + [refDb, fq1, fq2], stdout=aln_sam)
-        os.unlink(fq1)
-        os.unlink(fq2)
+            if min_score_to_filter:
+                # Filter reads in the alignment based on on their alignment score
+                aln_sam_filtered = util.file.mkstempfname('.sam')
```

Comment:
Output `.bam` by default as much as possible? Saving on disk IO typically makes up for compression/decompression CPU time.

Reply:
Great point. It should really be piped to `samtools` for output of a `.bam` intermediary, but this is older code we haven't gotten to yet. Maybe in a future PR where we made the change more broadly across the codebase. For now I just want to get bwa merged in...

Reply:
`samtools sort` works on piped input - it keeps memory usage down by doing an external sort on disk, so it can be sped up by option `-m` for memory usage